### PR TITLE
Arm support

### PR DESCRIPTION
**Pull request by udiz**
Addressing issue #35 - Adding support for arm64.

The following changes were made:
* Instead of a static vmlinux.h, we use a compact one download from an online source. Additional structures and constants are defined in a manual .h file.
* The target flag for clang was changed to compile the ebpf code to both amd64 and arm64.
* The build process has changed - it's all containerized now. The Dockerfile now has mult-stage building of itself using ebpf-builder as a first stage, running the `make` command in it.
* The github actions' docker build was modified to use both platforms, with QEMU set up.

### CODE REVIEW

File: pkg/tracing/ebpf/core_structures.h [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  TCP_CLOSE = 7,
+  TCP_CLOSE_WAIT = 8,
+  TCP_LAST_ACK = 9,
+  TCP_LISTEN = 10,
+  TCP_CLOSING = 11,
+  TCP_NEW_SYN_RECV = 12,
+  TCP_MAX_STATES = 13,
+};
+
+struct user_pt_regs {
```
Comment (avivzgroundcover) [inline pkg/tracing/ebpf/core_structures.h:39]: This is not a core structure - please move to an "arm_support.h" file

File: scripts/build/download_libbpf_headers.sh [hunk 2]
```diff
@@ -44,4 +44,13 @@ if [ "$?" -ne 0 ]; then
     exit 1
 fi
 
+# Fetch compact vmlinux file from cilium's ebpf repository.
+# This is not a libbpf header per-se, but it's close enough that we put it in the same location.
+curl -s -o "$HEADERS_DIRECTORY"/vmlinux.h \
+    https://raw.githubusercontent.com/cilium/ebpf/v${CILIUM_VMLINUX_VERSION}/examples/headers/common.h
```
Comment (avivzgroundcover) [inline scripts/build/download_libbpf_headers.sh:50]: Consider updating the cilium_vmlinux_version to v0.10.0 - latest release

**Review by avivzgroundcover — CHANGES_REQUESTED** [review]

File: scripts/build/download_libbpf_headers.sh (hunk 2 above)
Comment (udiz) [reply scripts/build/download_libbpf_headers.sh:50]: updated

File: pkg/tracing/ebpf/core_structures.h (hunk 1 above)
Comment (udiz) [reply pkg/tracing/ebpf/core_structures.h]: done

**Review by avivzgroundcover — APPROVED** [review]

**Pull request merged**